src/character.cpp

```diff
@@ -10447,15 +10447,22 @@ std::vector<Creature *> Character::get_visible_creatures( const int range ) cons
     } );
 }
 
-std::vector<Creature *> Character::get_targetable_creatures( const int range ) const
-{
-    return g->get_creatures_if( [this, range]( const Creature & critter ) -> bool {
-        bool can_see = sees( critter ) || sees_with_infrared( critter );
-        if( can_see )   //handles the case where we can see something with glass in the way or a mutation lets us see through walls
+std::vector<Creature *> Character::get_targetable_creatures( const int range, bool melee ) const
+{
+    std::vector<ter_id> glass = { t_door_glass_c, t_door_glass_o, t_door_glass_frosted_c, t_door_glass_frosted_o, t_wall_glass, t_wall_glass_alarm,
```

Comment:
We should jsonize it through terrain flag probably later.

Reply:
Yea either later tonight or tomorrow I'll take a stab at that.